SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -883,43 +897,40 @@ bool USpatialSender::SendRPC(const FPendingRPCParams& Params)
 		if (bCanPackRPC)
 		{
 			const UObject* UnresolvedObject = nullptr;
-			if (AddPendingRPC(TargetObject, Params, ComponentId, RPCInfo.Index, UnresolvedObject))
-			{
+			ERPCResult Result = AddPendingRPC(TargetObject, Function, Payload, ComponentId, RPCInfo.Index, UnresolvedObject);
 #if !UE_BUILD_SHIPPING
-				NetDriver->SpatialMetrics->TrackSentRPC(Function, RPCInfo.Type, Params.Payload.PayloadData.Num());
-#endif // !UE_BUILD_SHIPPING
-				return true;
-			}
-			else
+			if (Result == ERPCResult::Success)
 			{
-				return false;
+				NetDriver->SpatialMetrics->TrackSentRPC(Function, RPCInfo.Type, Payload.PayloadData.Num());
 			}
+			return Result;
+#endif // !UE_BUILD_SHIPPING
 		}
 		else
 		{
 			if (!NetDriver->StaticComponentView->HasAuthority(EntityId, ComponentId))
 			{
-				return false;
+				return ERPCResult::NoAuthority;
 			}
 
 			const UObject* UnresolvedParameter = nullptr;
-			Worker_ComponentUpdate ComponentUpdate = CreateRPCEventUpdate(TargetObject, Params.Payload, ComponentId, RPCInfo.Index, UnresolvedParameter);
+			Worker_ComponentUpdate ComponentUpdate = CreateRPCEventUpdate(TargetObject, Payload, ComponentId, RPCInfo.Index, UnresolvedParameter);
 
 			if (UnresolvedParameter)
 			{
-				return false;
+				return ERPCResult::UnresolvedParameters;
```

Comment:
This should never happen, so maybe a `verify`?

Reply:
Added ensure in corresponding functions, removed unused code.